src/preload.py

```diff
@@ -633,30 +633,43 @@ def find_non_empty_folder_in_path(path, child_dir=""):
 
 
 def find_docker_aufs_root(mountpoint):
-    # We're looking for a /docker/aufs/diff/<xxxxxxxxxxxxx>/ folder with some
+    # We're looking for a /<docker|balena>/aufs/diff/<xxxxxxxxxxxxx>/ folder with some
     # files not starting with a '.'
-    path = os.path.join(mountpoint, "docker", "aufs", "diff")
-    return find_non_empty_folder_in_path(path)
+    for name in ("docker", "balena"):
+        path = os.path.join(mountpoint, name)
+        if os.path.isdir(path):
+            path = os.path.join(path, "aufs", "diff")
+            return find_non_empty_folder_in_path(path)
 
 
 def find_docker_overlay2_root(mountpoint):
-    # We're looking for a /docker/overlay2/<xxxxxxxxxxxxx>/diff folder with
+    # We're looking for a /<docker|balena>/overlay2/<xxxxxxxxxxxxx>/diff folder with
     # some files not starting with a '.'
-    path = os.path.join(mountpoint, "docker", "overlay2")
-    return find_non_empty_folder_in_path(path, "diff")
+    for name in ("docker", "balena"):
+        path = os.path.join(mountpoint, name)
+        if os.path.isdir(path):
+            path = os.path.join(path, "overlay2")
+            return find_non_empty_folder_in_path(path, "diff")
 
 
-def get_docker_init_file_content(image=None):
+def get_docker_service_file_path(folder):
+    for name in ("docker", "balena"):
+        fpath = os.path.join(folder, "lib/systemd/system", name + ".service")
```

Comment:
This _could_ be
```
fpath = os.path.join(folder, "lib", "systemd", "system", name + ".service")
```

Reply:
Hmmm, if `find_docker_aufs_root` or `find_docker_overlay2_root` has already identified whether we're on a "docker system" or a "balena system", will we _ever_ have a mis-match of the two? I.e. could we ever have:
```
/docker/overlay2/<xxxxxxxxxxxxx>/diff/lib/systemd/system/balena.service
```
or
```
/balena/aufs/diff/<xxxxxxxxxxxxx>/lib/systemd/system/docker.service
```
?